### PR TITLE
Convert to composer lib with coverage tests

### PR DESCRIPTION
**Pull request by kabel**
API rewrite in order to utilize namespaces and be psr-4 autoload ready.

### CODE REVIEW

File: src/CachingService/CacheLite.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class CacheLite implements CachingServiceInterface
+{
+    const CACHE_NAMESPACE = 'ugbulletin';
+
+    protected $cache;
+
+    public function __construct()
+    {
+        if (!class_exists('Cache_Lite')) {
+            throw new \Exception('Unable to include Cache_Lite, is it installed?');
```
Comment (mfairchild365) [inline src/CachingService/CacheLite.php]: Nice error message!

Comment (kabel) [inline src/CachingService/CacheLite.php]: I hope that was a copy/paste.

Comment (kabel) [inline src/CachingService/CacheLite.php]: Whoo: https://github.com/unl/UNL_Services_CourseApproval/blob/a69b3e6c51ad55d5488d883776efe88cfacd6e7a/src/UNL/Services/CourseApproval/CachingService/CacheLite.php#L10

File: src/Course/Activities.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    public function count()
+    {
+        return count($this->xmlActivities);
+    }
+
+    public static function getFullDescription($activity)
+    {
+        $activities = static::getPossibleActivities();
+        if (!isset($activities[$activity])) {
+            throw new \Exception('Unknown activity type! '.$activity);
```
Comment (mfairchild365) [inline src/Course/Activities.php]: UnexpectedValueException?

File: src/Course/Activities.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    public function count()
+    {
+        return count($this->xmlActivities);
+    }
+
+    public static function getFullDescription($activity)
+    {
+        $activities = static::getPossibleActivities();
```
Comment (mfairchild365) [inline src/Course/Activities.php]: I had to look up `static::`... very cool!

File: src/Course/Activities.php (hunk 2 above)
Comment (kabel) [inline src/Course/Activities.php]: It certainly does make more sense: https://github.com/unl/UNL_Services_CourseApproval/blob/a69b3e6c51ad55d5488d883776efe88cfacd6e7a/src/UNL/Services/CourseApproval/Course/Activities.php#L50

File: src/Course/Activities.php (hunk 3 above)
Comment (kabel) [inline src/Course/Activities.php]: Late-static binding is, indeed, very cool.

**Pull request merged**